models/match.py

```diff
@@ -0,0 +1,50 @@
+"""Module containing the Match class and other related components"""
+
+from __future__ import annotations
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+from dataclasses import dataclass
+
+
+@dataclass
+class Match:
+    """A Premier League match between two teams in a particular season.
+
+    Instance Attributes:
+        - teams: A set of the the two teams names that played this match.
+        - date_time: The date and time of the match played.
+        - winner: The name of the winning team of the match, or None if the match ended in a draw.
+        - details: A mapping from each team name to its corresponding match details.
+
+    Representation Invariants:
+        - len(teams) == 2
+    """
+    teams: set[str]
+    date_time: datetime
+    details: dict[str, MatchDetails]
+    winner: Optional[str]
```

Comment:
The following representation invariants need to be added:
1. `winner` should be in teams
2. each name in teams should be in `details` dict

Reply:
Alright. Will add them.